### PR TITLE
Added fix for merge above on topmost cell to resolve #330

### LINKED ISSUE #330: merge with cell above merge with last cell
if you select the first cell of a notebook and merge with the cell above, it pulls the last cell of the notebook.

Probably because `cell(-1)` is the last... i doubt this was on purpose.

### PR DESCRIPTION
**Pull request by captainsafia**
Attempting to merge above on top-most cell will fail silently similar to merging down on bottom-most cell.

cc/ @Carreau

### CODE REVIEW

File: notebook/static/notebook/js/notebook.js [hunk 1]
```diff
@@ -1446,6 +1446,12 @@ define(function (require) {
         if (indices.length <= 1) {
             return;
         }
+
+        // Check if trying to merge above on topmost cell
+        if (indices.indexOf(-1) > -1) {
```
Comment (Carreau) [inline notebook/static/notebook/js/notebook.js]: Do we want to be more restrictive and check that no indices is negative ? (`.filter(function(value){return value < 0})`)

**Comment by Carreau** [comment]
+1 better fix than mine.

**Comment by ellisonbg** [comment]
Thanks Safia!

On Wed, Aug 26, 2015 at 7:28 AM, Matthias Bussonnier <
notifications@github.com> wrote:

> +1 better fix than mine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/340#issuecomment-135040464>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

File: notebook/tests/notebook/dualmode_merge.js [hunk 2]
```diff
@@ -67,6 +67,21 @@ casper.notebook_test(function () {
...
+    });
+    this.thenEvaluate(function() {
+        IPython.notebook.merge_cell_above();
+    });
+    this.then(function() {
+        this.test.assertEquals(this.get_cells_length(), 3, 'Merge cell 0 above: There are still 3 cells');
+        this.test.assertEquals(this.get_cell_text(0), a, 'Merge cell 0 above: Cell 0 is unchanged');
+        this.test.assertEquals(this.get_cell_text(1), b, 'Merge cell 0 above: Cell 1 is unchanged');
+        this.test.assertEquals(this.get_cell_text(2), c, 'Merge cell 0 above: Cell 2 is unchanged');
+        this.validate_notebook_state('merge up', command, 0);
```
Comment (Carreau) [inline notebook/tests/notebook/dualmode_merge.js]: I think you ment to quote `command` here.

File: notebook/static/notebook/js/notebook.js (hunk 1 above)
Comment (captainsafia) [inline notebook/static/notebook/js/notebook.js]: Hm. Yes. I wasn't anticipating any circumstances where an index value less
than negative one would be passed but I agree that adding a check for all
negative numbers would be better.

I'll add this as soon as I'm near a computer.

Comment (Carreau) [inline notebook/static/notebook/js/notebook.js]: No hurry.

File: notebook/tests/notebook/dualmode_merge.js (hunk 2 above)
Comment (captainsafia) [inline notebook/tests/notebook/dualmode_merge.js]: Nice catch!

Comment (Carreau) [inline notebook/tests/notebook/dualmode_merge.js]: Well not really, Travis was angry and all read complaining that `command` was undefined.

**Comment by captainsafia** [comment]
@Carreau — is this good to merge? It looks like there are some conflicts now...

**Comment by rgbkrk** [comment]
The conflicts are almost definitely related to us moving packaging around. If you're comfortable with a rebase, that's the best approach. I'd probably hold off for a little bit until we know for sure where to target this (we've changed the location of the JS in #268 which is why there's a conflict).

**Comment by captainsafia** [comment]
Mmmm. Yes. I see. I'll keep my eye posted on that PR until it is resolved.

**Comment by captainsafia** [comment]
I think this can be merged now that the revert has been executed.

cc/ @Carreau @rgbkrk

**Comment by rgbkrk** [comment]
This looks good and you added a test. Not much else I could ask for. :+1:

**Comment by Carreau** [comment]
Thanks !

**Pull request merged**